src/datachain/catalog/catalog.py

```diff
@@ -1751,3 +1751,41 @@ def index(
             client_config=client_config or self.client_config,
             only_index=True,
         )
+
+    def studio_dataset_versions(
```

Comment:
can it be abstracted somehow via metastore client or something? 

if not, should we move implementation into some studio related package? 

It feels studio stuff should not be part of the API (even a bit hidden one like catalog)

Reply:
The part to fetch the datasets from studio still lies in Studio Client. This is to parse the dataset and get the dataset versions or process it accordingly just similar to how we do the list for the `list_datasets_versions` . Since this serves the same purpose as the same. I think this should be here .

Reply:
Also, there is some logic to fetch the remote dataset and other.

Reply:
I still think we should not have `studio_...` named methods in API. How about `list_datasets_versions` accepting `studio=True` for example?

Reply:
That works for me.

Reply:
Updated accordingly.